### ISSUE TITLE
Ryuk v0.10.1 does not respond to SIGTERM signal from tc-java

### ISSUE THREAD

**Issue by Kehrlann** [testcontainers/moby-ryuk#168]
Using `tc-java` v1.19.7 ; when I run a simple test that runs one nginx container and then exits, Ryuk `v0.10.1` does not shut down. The test is the following:

```java
package testcontainers.repro;

import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.Test;
import org.testcontainers.containers.GenericContainer;

import java.net.URI;
import java.net.http.HttpClient;
import java.net.http.HttpRequest;
import java.net.http.HttpResponse;

class TestcontainersTest {
    private static final boolean DEBUG = false;

    @Test
    void test() throws Exception {
        try (GenericContainer<?> nginx = new GenericContainer<>("nginx:1-alpine-slim")) {
            nginx.withExposedPorts(80).start();
            HttpClient httpClient = HttpClient.newBuilder().build();
            String url = String.format("http://%s:%d", nginx.getHost(), nginx.getFirstMappedPort());
            HttpResponse<String> response = httpClient.send(HttpRequest.newBuilder().GET().uri(URI.create(url)).build(), HttpResponse.BodyHandlers.ofString());
            assertThat(response.statusCode()).isEqualTo(200);
            assertThat(response.body()).contains("<h1>Welcome to nginx!</h1>");
        }
    }
}

```

After the test is finished, Ryuk logs are as follows:

```
2024-10-11 10:56:01 time=2024-10-11T08:56:01.258Z level=INFO msg=starting connection_timeout=1m0s reconnection_timeout=10s request_timeout=10s shutdown_timeout=10m0s remove_retries=10 retry_offset=-1s port=8080 verbose=false
2024-10-11 10:56:01 time=2024-10-11T08:56:01.258Z level=INFO msg=Started address=[::]:8080
2024-10-11 10:56:01 time=2024-10-11T08:56:01.259Z level=INFO msg="client processing started"
2024-10-11 10:56:01 time=2024-10-11T08:56:01.298Z level=INFO msg="client connected" address=172.17.0.1:57760 clients=1
2024-10-11 10:56:01 time=2024-10-11T08:56:01.298Z level=INFO msg="adding filter" type=label values="[org.testcontainers=true org.testcontainers.lang=java org.testcontainers.version=1.19.8 org.testcontainers.sessionId=e4d8c29e-acf1-4de5-90a1-461a96630997]"
2024-10-11 10:56:01 time=2024-10-11T08:56:01.847Z level=INFO msg="signal received" clients=1 shutdown_timeout=10m0s
2024-10-11 10:56:01 time=2024-10-11T08:56:01.847Z level=INFO msg="client processing stopped"
2024-10-11 10:56:02 time=2024-10-11T08:56:02.221Z level=INFO msg="client disconnected" address=172.17.0.1:57760 clients=0
```

tc-java tries to turn off Ryuk using a JVM "shtudown hook", which kills Ryuk with SIGTERM - see [RyukResourceReaper](https://github.com/testcontainers/testcontainers-java/blob/94e4c8532a6f87e9b978f6c256681c41508ca2bc/core/src/main/java/org/testcontainers/utility/RyukResourceReaper.java#L80-L91)

In `v0.9.0`, Ryuk shuts down without issue. In `v0.10.1`, Ryuk does not take the SIGTERM signal into account. Trying to send it SIGTERM manually does not shut down Ryuk:

```
docker kill --signal SIGTERM testcontainers-ryuk-e4d8c29e-acf1-4de5-90a1-461a96630997
```

Note that sending SIGKILL does terminate Ryuk immediately.

**Comment by mdelapenya** [comment]
@Kehrlann we have just published `0.10.2`, which includes the fix. The GH runner will eventually publish the Docker image soon, so it'll be available in a couple of minutes.

Thanks!

**Comment by Kehrlann** [comment]
Just tested this, the image shuts down in approximately `10s` now 🎉

Thanks for the quick turnaround!
